vault/resource_approle_auth_backend_role.go

```diff
@@ -0,0 +1,384 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	approleAuthBackendRoleBackendFromPathRegex = regexp.MustCompile("^auth/(.+)/role/.+$")
+	approleAuthBackendRoleNameFromPathRegex    = regexp.MustCompile("^auth/.+/role/(.+)$")
+)
+
+func approleAuthBackendRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Create: approleAuthBackendRoleCreate,
+		Read:   approleAuthBackendRoleRead,
+		Update: approleAuthBackendRoleUpdate,
+		Delete: approleAuthBackendRoleDelete,
+		Exists: approleAuthBackendRoleExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"role_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the role.",
+				ForceNew:    true,
+			},
+			"role_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The RoleID of the role. Autogenerated if not set.",
+			},
+			"bind_secret_id": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Whether or not to require secret_id to be present when logging in using this AppRole.",
+			},
+			"bound_cidr_list": {
```

Comment:
Is this list guaranteed to be sorted, or does it need to be sorted? `TypeList` is great for things that should maintain an order

Reply:
As far as I know, it's not guaranteed to be sorted, but I can add sort to it before setting it in state if we're worried about it.

Reply:
Oops, no, that's a bad idea. I'll just make it a Set :)